source/Sgghost.h

```diff
@@ -0,0 +1,139 @@
+#ifndef SGGHOST_H
+#define SGGHOST_H
+
+// #include "../../Empirical/include/emp/math/Random.hpp"
+// #include "../../Empirical/include/emp/tools/string_utils.hpp"
+// #include <set>
+// #include <iomanip> // setprecision
+// #include <sstream> // stringstream
+#include "Host.h"
+#include "SymWorld.h"
+
+
+class sggHost: public Host {
+protected:
+  double sourcepool=0;
+
+
+public:
+  sggHost(emp::Ptr<emp::Random> _random, emp::Ptr<SymWorld> _world, emp::Ptr<SymConfigBase> _config,
+  double _intval =0.0, emp::vector<emp::Ptr<Organism>> _syms = {},
+  emp::vector<emp::Ptr<Organism>> _repro_syms = {},
+  std::set<int> _set = std::set<int>(),
+  double _points = 0.0) : Host(_random, _world, _config, _intval,_syms, _repro_syms, _set, _points) { 
+    if ( _intval > 1 || _intval < -1) {
+       throw "Invalid interaction value. Must be between -1 and 1";  // Exception for invalid interaction value
+     };
+   }
+
+
+
+  sggHost(const sggHost &) = default;
+  sggHost(sggHost &&) = default;
+  sggHost() = default;
+ 
+  double GetPool() {return sourcepool;}
+  void SetPool(double _in) {sourcepool= _in;}
+  void AddPool(double _in) {sourcepool += _in;}
+
+
+  void DistribResources(double resources) {
+    double hostIntVal = interaction_val; //using private variable because we can
+
+    //In the event that the host has no symbionts, the host gets all resources not allocated to defense or given to absent partner.
+    if(syms.empty()) {
+
+      if(hostIntVal >= 0){
+	      double spent = resources * hostIntVal;
+        this->AddPoints(resources - spent);
+      }
+      else {
+        double hostDefense = -1.0 * hostIntVal * resources;
+        this->AddPoints(resources - hostDefense);
+      }
+      return; //This concludes resource distribution for a host without symbionts
+    }
+
+    //Otherwise, split resources into equal chunks for each symbiont
+    int num_sym = syms.size();
+    double hostDonation = 0;
+    double sym_piece = (double) resources / num_sym;
+   
+
+    for(size_t i=0; i < syms.size(); i++){
+      if(hostIntVal < 0){
+        double hostDefense = hostIntVal * sym_piece * -1.0;
+        hostDonation = 0;
+        SetResInProcess(sym_piece - hostDefense);
+      }
+      else if(hostIntVal >= 0){
+        hostDonation = hostIntVal * sym_piece;
+        SetResInProcess(sym_piece - hostDonation);
+      }
+      double sym_return = syms[i]->ProcessResources(hostDonation);
+      this->AddPoints(sym_return + GetResInProcess());
+      SetResInProcess(0);
+      double hostPool = syms[i]->ProcessPool();
+      this->AddPool(hostPool);
+    }
+    if(syms.size()>0){this->DistribPool();}
+    
+
+  } //end DistribResources
+
+  void DistribPool(){
+    //to do: marginal return
+    int num_sym = syms.size();
+    double sym_piece = (double) sourcepool / num_sym;
+    for(size_t i=0; i < syms.size(); i++){
+        syms[i]->AddPoints(sym_piece);
+    }
+    this->SetPool(0);
+  }
+
+  void Process(size_t location) {
```

Comment:
Looks like this function is getting run by any of the tests. Try to add at least one that runs it.